environment-pipelines/tests/unit.tftest.hcl

```diff
@@ -695,6 +761,203 @@ run "test_iam" {
+    error_message = "Input artifacts incorrect"
+  }
+
+  assert {
+    condition     = aws_codepipeline.environment_pipeline.stage[7].action[0].input_artifacts[0] == "build_output"
+    error_message = "Input artifacts incorrect"
+  }
+
+  assert {
+    condition     = aws_codepipeline.environment_pipeline.stage[7].action[0].configuration.EnvironmentVariables == "[{\"name\":\"TRIGGERED_ACCOUNT_ROLE_ARN\",\"value\":\"arn:aws:iam::000123456789:role/my-app-triggered-pipeline-trigger-pipeline-from-my-pipeline\"},{\"name\":\"TRIGGERED_PIPELINE_NAME\",\"value\":\"my-app-triggered-pipeline-environment-pipeline\"},{\"name\":\"TRIGGERED_PIPELINE_AWS_PROFILE\",\"value\":\"prod\"},{\"name\":\"SLACK_THREAD_ID\",\"value\":\"#{variables.SLACK_THREAD_ID}\"},{\"name\":\"SLACK_CHANNEL_ID\",\"type\":\"PARAMETER_STORE\",\"value\":\"/codebuild/slack_pipeline_notifications_channel\"},{\"name\":\"SLACK_REF\",\"value\":\"#{slack.SLACK_REF}\"}]"
+    error_message = "Configuration Env Vars incorrect"
+  }
+
+  assert {
+    condition     = aws_codepipeline.environment_pipeline.stage[7].action[0].configuration.ProjectName == "my-app-my-pipeline-environment-pipeline-trigger"
+    error_message = "Configuration ProjectName incorrect"
+  }
+
+  assert {
+    condition     = aws_codepipeline.environment_pipeline.stage[7].action[0].configuration.PrimarySource == "build_output"
+    error_message = "Configuration PrimarySource incorrect"
+  }
+
+  assert {
+    condition     = aws_codepipeline.environment_pipeline.stage[7].action[0].namespace == null
+    error_message = "Namespace incorrect"
+  }
+}
+
+run "test_triggered_pipelines" {
+  command = plan
+
+  variables {
+    pipeline_name = "triggered-pipeline"
+  }
+
+  assert {
+    condition     = local.triggers_another_pipeline == false
+    error_message = ""
+  }
+
+  assert {
+    condition     = aws_iam_role.trigger_pipeline["my-pipeline"].name == "my-app-triggered-pipeline-trigger-pipeline-from-my-pipeline"
+    error_message = ""
+  }
+
+  assert {
+    condition     = aws_iam_role.trigger_pipeline["my-pipeline"].assume_role_policy == "{\"Sid\": \"AssumeTriggerCodePipeline\"}"
+    error_message = ""
+  }
+
+  assert {
+    condition     = jsonencode(aws_iam_role.trigger_pipeline["my-pipeline"].tags) == jsonencode(var.expected_tags)
+    error_message = ""
+  }
+
+  assert {
+    condition     = aws_iam_role_policy.trigger_pipeline["my-pipeline"].name == "my-app-triggered-pipeline-trigger-pipeline-from-my-pipeline"
+    error_message = ""
+  }
+
+  assert {
+    condition     = aws_iam_role_policy.trigger_pipeline["my-pipeline"].role == "my-app-triggered-pipeline-trigger-pipeline-from-my-pipeline"
+    error_message = ""
+  }
+
+  # aws_iam_role_policy.trigger_pipeline["my-pipeline"].policy cannot be tested on a plan
+
+  assert {
+    condition     = length(toset(local.list_of_triggering_pipelines)) == 1
+    error_message = ""
+  }
```

Comment:
I wonder if we can have a few more tests around the new locals? There's quite a bit of logic in them. At least the ones the are used outside of locals.tf. 

This one for example isn't used outside of locals.tf I don't think, but `set_of_triggering_pipelines` is, and also `triggered_pipeline_account_role`

Be better to assert against the actual content as well rather than just the length.

Reply:
added three tests to cover these locals, checking for more

Reply:
these were all the locals we needed to cover (the 3 you mentioned)